ringbuffer.h

```diff
@@ -0,0 +1,24 @@
+#ifndef M8C_RINGBUFFER_H
+#define M8C_RINGBUFFER_H
+
+#include <stdlib.h>
+
```

Comment:
I unfortunately couldn't compile the code with cmake configuration, I think stdint.h should be included in this file and the respective .c file too to get uint32_t etc.?